addons/hr_calendar/models/calendar_event.py

```diff
@@ -0,0 +1,55 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from pytz import UTC
+from odoo import api, fields, models
+from odoo.addons.resource.models.utils import Intervals, sum_intervals, timezone_datetime
+
+
+class CalendarEvent(models.Model):
+    _inherit = "calendar.event"
+
+    unavailable_partner_ids = fields.Many2many('res.partner', compute='_compute_unavailable_partner_ids')
+
+    def _get_events_interval(self):
+        """
+        Calculate the interval of an event based on its start, stop, and allday values. If an event is scheduled for the
+        entire day, its interval will correspond to the work interval defined by the company's calendar.
+        """
+        start = min(self.mapped('start')).replace(hour=0, minute=0, second=0, tzinfo=UTC)
+        stop = max(self.mapped('stop')).replace(hour=23, minute=59, second=59, tzinfo=UTC)
+        if not start or not stop:
+            return {}
```

Comment:
What do you think? Imagine start or stop is not provided, if so we do first two operations for no reason
```suggestion
        if not start or not stop:
            return {}
        start = min(self.mapped('start')).replace(hour=0, minute=0, second=0, tzinfo=UTC)
        stop = max(self.mapped('stop')).replace(hour=23, minute=59, second=59, tzinfo=UTC)
```

Reply:
Start and stop are not given in parameter; I will take the minimum start and the maximum stop. So I will go in this loop only if all events don't have a start or stop date :)

Reply:
correct ^^